### ISSUE TITLE
UBSDKRideRequestButtonDelegate methods not invoked.

### ISSUE THREAD

**Issue by k28gupta** [uber/rides-ios-sdk#76]
This is the Ride request button:    

```
UBSDKRideRequestViewRequestingBehavior *requestBehavior = [[UBSDKRideRequestViewRequestingBehavior alloc] initWithPresentingViewController:self];

requestBehavior.modalRideRequestViewController.delegate = self;
UBSDKRideParametersBuilder *builder = [[UBSDKRideParametersBuilder alloc] init];
[builder setPickupToCurrentLocation];
UBSDKRideParameters *parameters = [builder build];
UBSDKRideRequestButton *button = [[UBSDKRideRequestButton alloc] initWithRideParameters: parameters requestingBehavior: requestBehavior];

button.delegate = self;
```

I am receiving the callbacks for UBSDKModalViewControllerDelegate Methods, but no callbacks when a ride is accepted or if an error occurs.

**Comment by jbrophy17** [comment]
Currently, there are no callbacks to indicate when a ride is accepted (see #53). As for error handling, most of that is taken care of for you by the Ride Request View Controller. Are you trying to add custom error handling logic?

**Comment by k28gupta** [comment]
@jbrophy17  But what is the point of this delegate then "UBSDKRideRequestButtonDelegate" which is defined in the RideRequestButton... It has two methods which are not getting invoked. So, is this a known issue or am I doing something wrong?
1. func rideRequestButtonDidLoadRideInformation(button: RideRequestButton)
2. func rideRequestButton(button: RideRequestButton, didReceiveError error: RidesError)

**Comment by jbrophy17** [comment]
`func rideRequestButtonDidLoadRideInformation(button: RideRequestButton)` is called when the button loads the information based on the ride parameters you set. So when it updates to display the product name and the estimated price

`func rideRequestButton(button: RideRequestButton, didReceiveError error: RidesError)` is called if the button encounters an error when loading the information.

**Comment by PaulTaykalo** [comment]
@jbrophy17 here are some cases, when the button doesn't have *any* delegate methods:
https://github.com/uber/rides-ios-sdk/blob/master/source/UberRides/RideRequestButton.swift#L423

And then

https://github.com/uber/rides-ios-sdk/blob/master/source/UberRides/RideRequestButton.swift#L378

So in this case, we won't receive any information about time estimations, and we, as developers won't know, if there'll be any updates any time soon for the button, or may be we should trigger another one with new location.

**Comment by jbrophy17** [comment]
@PaulTaykalo That is a good find, if the time estimate response is empty, then no delegate method will be called. If you like, you can put up a PR and I would be happy to review it, otherwise I will patch this for the next update

**Comment by futuretap** [comment]
In case of `[builder setPickupToCurrentLocation]` the delegate is not called as well because the guard in RideRequestButton.swift:368 causes an early exit without error callback.

**Comment by jbrophy17** [comment]
@futuretap also a great find. I will make sure the fix also covers this use case

**Comment by jbrophy17** [comment]
Fixed in [v0.5.3](https://github.com/uber/rides-ios-sdk/releases/tag/v0.5.3)
